core/controllers/classroom.py

```diff
@@ -29,13 +29,24 @@ class ClassroomPage(base.BaseHandler):
     """Renders the classroom page."""
 
     @acl_decorators.open_access
-    def get(self):
+    def get(self, classroom_name):
         """Handles GET requests."""
 
         if not constants.ENABLE_NEW_STRUCTURE_PLAYERS:
             raise self.PageNotFoundException
 
-        self.render_template('classroom-page.mainpage.html')
+        # If classroom name in the URL is /Math or /mAtH, redirect it to /math.
+        # If classroom name is already all lower cased, render the appropriate
+        # classroom page.
+        for classroom_dict in config_domain.TOPIC_IDS_FOR_CLASSROOM_PAGES.value:
+            if classroom_dict['name'].lower() == classroom_name:
+                self.render_template('classroom-page.mainpage.html')
+                break
+            if classroom_dict['name'].lower() == classroom_name.lower():
+                self.redirect('/%s' % (classroom_name.lower()))
+                break
```

Comment:
Instead of breaking at this point, return instead. (In case code gets added after this in the future.)

Reply:
Obsolete due to recent changes.